### ISSUE TITLE
Notification tray icons are deprecated in gnome-shell; recommend TopIcons extension

### ISSUE THREAD

**Issue by ryneches** [keybase/client#5772]
The notification area / message tray / thingy [was deprecated in gnome-shell about six years ago](https://wiki.gnome.org/Design/OS/MessageTray/Compatibility), and the legacy implementation is [broken and horrible](https://www.reddit.com/r/linux/comments/32byhw/gnome_316_remove_message_tray_on_bottom_left_of/). This likely a contributing factor in #2710 #3490 #4110.

In lieu of wading into the tawdry debate among the Gnome and Unity designers and attempting to implement a solution, I recommend adding a note in the installation instructions alerting Gnome users of the availability of the [TopIcons extension](https://extensions.gnome.org/extension/1031/topicons/), which basically makes the whole problem go away. It's a violation of some of the Gnome UX/UI design principles I guess, but it doesn't look any uglier than MacOS.

**Comment by oconnor663** [comment]
Yes, I've been using the topicons plugin ever since I installed Gnome 3. Even before Keybase had a tray icon, I needed it for Dropbox. Where do you suggest we put a mention? Somewhere on [this page](https://keybase.io/docs/the_app/install_linux)?

**Comment by ryneches** [comment]
Ooops. Sorry for letting this sit. Yes, I think that would be a fine place to point people to TopIcons.

I do wonder if this problem is going to get some attention from the gnome-shell team now that Ubuntu is adopting Gnome.

**Comment by jalberto** [comment]
@ryneches topicons extension is not maintained nor working in latest stable gnome.
There are some alternatives (at different degree of "working"). Gnome team decided to get rid of those status icons (https://blogs.gnome.org/aday/2017/08/31/status-icons-and-gnome/) and with fair reasons behind.

I suggest a separate gnome-extension exclusively for keybase can improve usability

**Comment by oconnor663** [comment]
@jalberto what is Dropbox going to do for Gnome. As far as the tray icon is concerned, I assume we'll want to do the same thing.

**Comment by jalberto** [comment]
@oconnor663 I don't use dropbox so I don't know, but whatever decision they take, is not necessarily the best option but the mots convenient for them.

anyway seems like keyba status icon implementation is not working 100%, as using this extension by ubuntu https://extensions.gnome.org/extension/615/appindicator-support/ works for discord and skype but not for keybase

I actually agree with Gnome decision of deprecating statsu icons as specs were never 100% standard or flexible enough, but that is a different topic.

**Comment by oconnor663** [comment]
My current workaround is to run Gnone Shell in legacy X11 mode, where TopIcons still works. For an actual solution, we might need to wait until Electron adds official support for libappindicator. (This seems to be [under debate](https://github.com/electron/electron/issues/10427).)

> a separate gnome-extension exclusively for keybase

I'd be willing to review an open source submission if anyone wanted to try this. Otherwise, it's unlikely we'll want to implement something specific for Gnome.

**Comment by MarkoPaasila** [comment]
How can I open keybase-gui main window from command line? I suspect it starts minimized, which means it's inaccessible in gnome.

**Comment by jalberto** [comment]
I wonder if keybase use in Linux is so low then the team don't think it worth to invest time improving the experience. Or maybe every KB linux user is using KDE?

This bug is open for 2 years now, with no solution or proposal from KB team

**Comment by MarkoPaasila** [comment]
Should I make a new issue with "Can't open keybase on Gnome any more"?

**Comment by FilBot3** [comment]
TopIcons Plus works with Discord, Teams, and Slack, but not keybase. 
* https://extensions.gnome.org/extension/1031/topicons/

Is there an extension that works for this, or just get over it or go back to KDE?

**Comment by erdii** [comment]
I may be a little late to the party but since gnome 40 I use:
https://extensions.gnome.org/extension/615/appindicator-support/
and
`libappindicator-gtk3` <- this is the library package name on arch linux
and they work perfectly.

![image](https://user-images.githubusercontent.com/5539202/116523531-9e770500-a8d6-11eb-9c2a-028f38164cc9.png)

**Comment by pasqualirb** [comment]
For GNOME 41, see [Tray Icons Reloaded](https://extensions.gnome.org/extension/2890/tray-icons-reloaded) ([GitHub repo](https://github.com/MartinPL/Tray-Icons-Reloaded)).
